src/stratigraphy/util/layer_identifier_column.py

```diff
@@ -92,6 +116,50 @@ def is_contained(self, rect: fitz.Rect) -> bool:
             and self.rect().y1 <= rect.y1
         )
 
+    def get_depth_interval(self, block: TextBlock) -> LayerDepthColumnEntry:
+        """Extract depth interval from a material description block.
+
+        For borehole profiles in the Deriaz layout, the depth interval is usually found in the text description
+        of the material. Often, these text descriptions contain a further separation into multiple sub layers.
+        These sub layers have their own depth intervals. This function extracts the overall depth interval,
+        spanning across all mentioned sub layers.
+
+        Args:
+            block (TextBlock): The block to calculate the depth interval for.
+
+        Returns:
+            LayerDepthColumnEntry: The depth interval.
+        """
+        depth_entries = []
+        for line in block.lines:
+            try:
+                new_entries = extract_layer_depth_interval_entries(line.text, line.rect, require_start_of_string=False)
+                # require_start_of_string = False because the depth interval may not always start at the beginning
+                # of the line e.g. "Remblais Heterogene: 0.00 - 0.5m"
+                if new_entries:
+                    depth_entries.append(new_entries)
+            except ValueError:
+                pass
+
+        if depth_entries:
+            # Merge the sub layers into one depth interval.
+            start_idx = np.argmin([entry[0].value for entry in depth_entries])
+            end_idx = np.argmax([entry[1].value for entry in depth_entries])
+
+            start = depth_entries[start_idx][0]
+            end = depth_entries[end_idx][1]
```

Comment:
isn't this equivalent to
```
            start = min([entry[0].value for entry in depth_entries])
            end = max([entry[1].value for entry in depth_entries])
```
?

Reply:
Yes that's correct. The complexity stems from an older implementation with two lists.